k9mail/src/main/java/com/fsck/k9/helper/Utility.java

```diff
@@ -459,6 +472,45 @@ public static String extractMessageId(final String text) {
         return null;
     }
 
+    public static String getResizedImageFile(Context context, Uri uri, float multiplier){
+        File cacheDir = context.getCacheDir();
+        File tempAttachmentsDirectory = new File(cacheDir.getPath() + "/tempAttachments/");
+        tempAttachmentsDirectory.mkdirs();
+        final File tempFile = new File(tempAttachmentsDirectory, String.valueOf(System.currentTimeMillis()) + ".tmp");
+
+        Bitmap bitmap = null;
+        Bitmap resized = null;
+        try {
+            bitmap = MediaStore.Images.Media.getBitmap(context.getContentResolver(), uri);
+            resized = Bitmap.createScaledBitmap(bitmap, (int) (bitmap.getWidth() * multiplier), (int) (bitmap.getHeight() * multiplier), true);
+            FileOutputStream out = new FileOutputStream(tempFile);
+            resized.compress(Bitmap.CompressFormat.PNG, 100, out);
+            out.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "";
+        }
+
+        return tempFile.getAbsolutePath();
+    }
+
+    public static void clearTemporaryAttachmentsCache(Context context){
+        File cacheDir = context.getCacheDir();
+        File tempAttachmentsDirectory = new File(cacheDir.getPath() + "/tempAttachments/");
+        if(tempAttachmentsDirectory.exists()){
+            try {
+                FileUtils.cleanDirectory(tempAttachmentsDirectory);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Unhandled error case. Again, we have code for temporary file storage. So use that.